src/neuroglia/data/queries/queryable.py

```diff
@@ -110,18 +112,10 @@ def _get_lambda_source_code(self, function : Callable):
         if len(source_lines) != 1: return None
         source_text = os.linesep.join(source_lines).strip()
         source_ast = ast.parse(source_text)
-        lambda_node = next((node for node in ast.walk(source_ast) if isinstance(node, ast.Lambda)), None)
+        lambda_node = next((node for node in ast.walk(source_ast) if isinstance(node, ast.Lambda) and node.col_offset <= max_col_offset), None)
         if lambda_node is None: return None
-        lambda_text = source_text[lambda_node.col_offset:]
-        lambda_body_text = source_text[lambda_node.body.col_offset:]
-        min_length = len('lambda:_')
-        while len(lambda_text) > min_length:
-            try:
-                compile(lambda_body_text, '<unused filename>', 'eval')
-                return lambda_text
-            except SyntaxError:
-                lambda_text = lambda_text[:-1]
-                lambda_body_text = lambda_body_text[:-1]
```

Comment:
I'm not sure why the `while-try-except` instead of using `end_col_offset`, is there a specific reason/use-case I missed ?

Reply:
Nope, AFAIK your way is far better!